packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx

```diff
@@ -54,7 +54,7 @@ export const OnboardingProvider = ({
   );
 
   const toggleStepComplete = (step: OnboardingStep) => {
-    const isComplete = !stepStatus[step.id];
+    const isComplete = !(stepStatus[step.id] ?? false);
```

Comment:
yeah, you're right, I'll rethink naming so we don't have to cast opposite boolean values.
stepStatus[step.id] could be boolean, but now - since one of the steps is optional it can be undefined as well, and we want isComplete to have a boolean value

Reply:
But when you do `!undefined` it's the same as `!(undefined ?? false)`

Reply:
Shitty rule haha

`!Boolean(<var>)` should make it work

Reply:
yeah, it does! 
i thought !!(value) does the same what Boolean(value) and only checked the first option -but I was wrong
changed with your suggestion